broker/l2tp_broker.py

```diff
@@ -59,7 +59,7 @@
   # Message data (with length prefix)
   cs.PascalString("data"),
   # Pad the message so it is at least 12 bytes long
-  cs.Padding(lambda ctx: max(0, 6 - len(ctx["data"]))),
+  cs.Optional(cs.Padding(lambda ctx: max(0, 6 - len(ctx["data"])))),
```

Comment:
Can you also remove the code around line 525 (method `Tunnel._run`) that adds padding before parsing if there is none, as now this is probably not necessary and your fix is actually much nicer.